tests/field_defaults/tests.py

```diff
@@ -14,3 +33,217 @@ def test_field_defaults(self):
         self.assertIsInstance(a.id, int)
         self.assertEqual(a.headline, "Default headline")
         self.assertLess((now - a.pub_date).seconds, 5)
+
+    @skipUnlessDBFeature("can_return_columns_from_insert")
+    def test_field_database_defaults_returning(self):
+        a = DBArticle()
+        now = datetime.now()
+        a.save()
+
+        if connection.vendor == "sqlite":
+            # Round microseconds down to the nearest millisecond
+            microseconds = (now.microsecond // 1000) * 1000
+            now = now.replace(microsecond=microseconds)
+
+        self.assertIsInstance(a.id, int)
+        self.assertEqual(a.headline, "Default headline")
+
+        # Work around a flakiness in the Oracle setup
+        if connection.vendor == "oracle" and now > a.pub_date:
+            self.assertLess((now - a.pub_date).seconds, 5)
+        else:
+            self.assertLess((a.pub_date - now).seconds, 5)
+
+    @skipIfDBFeature("can_return_columns_from_insert")
+    def test_field_database_defaults_refresh(self):
+        a = DBArticle()
+        now = datetime.utcnow()
+        a.save()
+        a.refresh_from_db()
+
+        precision = connection.features.time_cast_precision
+        if precision < 6:
+            precision = 10 ** (6 - precision)
+            # Truncate to milliseconds on SQLite
+            microsecond = now.microsecond // precision * precision
+            now = now.replace(microsecond=microsecond)
+
+        self.assertIsInstance(a.id, int)
+        self.assertEqual(a.headline, "Default headline")
+        self.assertLess((a.pub_date - now).seconds, 5)
+
+    def test_bulk_create_all_db_defaults(self):
+        articles = [DBArticle(), DBArticle()]
+        DBArticle.objects.bulk_create(articles)
+
+        expected_headlines = ["Default headline", "Default headline"]
+        headlines = DBArticle.objects.values_list("headline", flat=True)
+        self.assertCountEqual(headlines, expected_headlines)
+
+    def test_bulk_create_all_db_defaults_one_field(self):
+        articles = [
+            DBArticle(pub_date=datetime.now()),
+            DBArticle(pub_date=datetime.now()),
+        ]
+        DBArticle.objects.bulk_create(articles)
+
+        expected_headlines = ["Default headline", "Default headline"]
+        headlines = DBArticle.objects.values_list("headline", flat=True)
+        self.assertCountEqual(headlines, expected_headlines)
+
+    def test_bulk_create_mixed_db_defaults(self):
+        articles = [DBArticle(), DBArticle(headline="Something else")]
+        DBArticle.objects.bulk_create(articles)
+
+        expected_headlines = ["Default headline", "Something else"]
+        headlines = DBArticle.objects.values_list("headline", flat=True)
+        self.assertCountEqual(headlines, expected_headlines)
+
+    @skipUnlessDBFeature(
+        "supports_expression_defaults",
+        "MySQL before 8.0.13 does not support function defaults.",
+    )
```

Comment:
IMO, this comment is unnecessary:
```suggestion
    @skipUnlessDBFeature("supports_expression_defaults")
```